### DISCUSSION TITLE
suggestion: reorginaztion of user_setups folder

### DISCUSSION THREAD

**Discussion by ECO1AI** [Bodmer/TFT_eSPI#3205] [New processor/display support or capability request]
this is a suggestion to rearrange `user_setups` folder within the library as follows:

```
user_setups:
                      list_by_driver
                                             list_by_LCD
                                                                list_by_board
                                                                                       config_file
                                                                board_pic.png
                                             note.txt or note.md
```

example:
```
user_setups
             |----- ST7796
                            |---- spi_revA
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_revB
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_parallel_8
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_parallel_16
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |---- spi_revA_pic.png
                            |----spi_revB_pic.png
                            |----spi_parallel_8_pic.png
                            |----spi_parallel_16_pic.png
             |----- ILI8894
                            |---- spi_revA
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_revB
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_parallel_8
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |----spi_parallel_16
                                           |---- Arduino
                                           |---- RPI_2040
                                           |---- ESP32
                            |---- spi_revA_pic.png
                            |----spi_revB_pic.png
                            |----spi_parallel_8_pic.png
                            |----spi_parallel_16_pic.png
```
note:

- if there are two versions of the same LCD, we indicate them by `revA` , `revB`.
- if there is no ESP driver for example, we remove the folder.
- `note.txt` or `note.md` are are to indicate the nonexistence of driver support for a board or to describe the limitation of the driver, as example.
- picture are important to illustrate the LCD and differentiate between revs.

as such, there will be some changes to `user_setup_select.h` file:

instead of listing all combinations we aim for the following:
```
#defined LCD_DRIVER
#defined LCD_REV
#defined CONTROLLER_BOARD
```
this way, the library will follow `/LCD_DRIVER/LCD_REV/CONTROLLER_BOARD/config.h` to find the needed driver. as example:`/ST7796/SPI_REVA/ESP32/config.h`

As `tft_espi` library is getting larger and larger, its getting hard to keep track and find the target driver for our experiment. i am ware that it needs sometime and effort to make this change,but slowly we can convert to the new cofig format.

**Comment by Bodmer** [comment]
You can create your own setup folder. See here:
https://github.com/Bodmer/TFT_eSPI#tips

There are no pans to change TFT_eSPI as I think this will create problems for existing users familiar with the current setup arrangement.

**Reply by ECO1AI** [reply]
> this will create problems for existing users familiar with the current setup

i understand your point.

but how about the new users?

also, the more LCD's and board added, the harder it is to find what you need; because they are all in one folder.

how about i create the now config file(will take time, and there might be some errors due to lack of info's), then give the user a choice to use any of the two config.

after the new config is merge and you add a config file, mention me (in the discussion or issue) so i add it to the new config.
